[wasm] Do not use link flags for bitcode compilation

radekdoulik opened the pull request:
This avoids errors like:

    /root/helix/work/correlation/build/wasm-shared/WasmApp.Common.targets(832,5): error : emcc: warning: linker setting ignored during compilation: 'EXPORT_ES6' [-Wunused-command-line-argument] [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
    /root/helix/work/correlation/build/wasm-shared/WasmApp.Common.targets(832,5): error : emcc: warning: linker setting ignored during compilation: 'EXPORT_EXCEPTION_HANDLING_HELPERS' [-Wunused-command-line-argument] [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]

radekdoulik committed:
[wasm] Do not use link flags for bitcode compilation

This avoids errors like:

    /root/helix/work/correlation/build/wasm-shared/WasmApp.Common.targets(832,5): error : emcc: warning: linker setting ignored during compilation: 'EXPORT_ES6' [-Wunused-command-line-argument] [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
    /root/helix/work/correlation/build/wasm-shared/WasmApp.Common.targets(832,5): error : emcc: warning: linker setting ignored during compilation: 'EXPORT_EXCEPTION_HANDLING_HELPERS' [-Wunused-command-line-argument] [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]

akoeplinger commented on src/mono/wasm/build/WasmApp.Common.targets:
```diff
@@ -804,7 +804,7 @@
   -->
   <Target Name="_WasmWriteRspForCompilingBitcode">
-    <WriteLinesToFile Lines="@(_BitcodeLDFlags)" File="$(_WasmCompileBitcodeRsp)" Overwrite="true" WriteOnlyWhenDifferent="true" />
+    <WriteLinesToFile Lines="@(_EmccCommonFlags)" File="$(_WasmCompileBitcodeRsp)" Overwrite="true" WriteOnlyWhenDifferent="true" />
```
the comment above still references `_BitcodeLDFlags`.

I'm also lacking enough context to check why `_EmccCommonFlags` is enough 😄

radekdoulik replied:
updated the comment. also thinking more about it, I replaced the original _BitcodeLDFlags with _BitcodeCompileFlags. that way it is more clear and we still provide way to inject extra flags for BC compilation

radekdoulik replied:
the previously used `_EmccLDFlags` are based on `_EmccCommonFlags` plus linking specific flags. similar for the `_WasiClangLDFlags`

radekdoulik committed:
Rename _BitcodeLDFlags to _BitcodeCompileFlags

Update the comment as well.

_BitcodeLDFlags were used only for the BC compilation. This way
we keep the possibility to add extra flags via EmccExtraBitcodeCompilationFlags and
WasiClangExtraBitcodeCompileFlags.

radekdoulik committed:
Merge branch 'main' into pr-wasm-fix-bc-compilation

radekdoulik committed:
Merge branch 'main' into pr-wasm-fix-bc-compilation

kg commented:
This seems like it may have caused an increase in file size for AOT and a resulting regression in AOT startup time. The regression is bigger for cold start than warm start, but both of them got worse around the time this landed

radekdoulik commented:
/backport to release/8.0
